pyproject.toml

```diff
@@ -63,4 +60,47 @@ dev = [
     "pre-commit~=4.3",
     "pytest~=8.4",
     "pytest-cov~=7.0",
+    "ruff==0.13.1",  # hard pinned to match the pre-commit-config
+]
+
+[tool.ruff]
+line-length = 120
+# gitignore contents (and potentially other ignore files) are also excluded
+respect-gitignore = true
+# overlaps with existing gitignore exclusions are ok
+extend-exclude = [
+    "**/env/**",
+    "**/build/**",
+    "**/dist/**",
+    "**/_build/**",
+    "**/.eggs/**",
+    "**/.git/**",
+    "**/.hg/**",
+    "**/.mypy_cache/**",
+    "**/.nox/**",
+    "**/.venv/**",
+    "**/docs/**",
+]
+
+[tool.ruff.lint]
+select = ["ALL"]
+ignore = [
+    "COM",  # flake8 comma rules - redundant with ruff format
+    "D",  # pydocstyle - not currently a priority
+    "PLR2004",  # Magic value used in comparison - not currently a priority
+    "FIX",  # comments with follow up - not currently a priority
+    "TD",  # comments with follow up - not currently a priority
+    # TODO: renable the following
```

Comment:
I think a few of these are useful and important but not critical enough to block anything currently.